File: ts/src/types/marketplace.ts
export type OrderStatus = "pending" | "confirmed" | "shipped" | "delivered" | "cancelled"

export interface Product {
  id: string
  name: string
  description?: string
  price: number
  category?: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface Order {
  id: string
  product_id: string
  user_id: string
  status: OrderStatus
  quantity: number
  total_price: number
  created_at: string
  updated_at: string
}

export interface CreateOrderInput {
  product_id: string
  quantity: number
  metadata?: Record<string, unknown>
}

File: py/src/imbrace/types/marketplace.py
from __future__ import annotations
from typing import Any, Dict, List, Literal, Optional
from pydantic import BaseModel


class ProductVariant(BaseModel):
    id: str
    name: str
    price: float
    stock: int
    attributes: Optional[Dict[str, Any]] = None


class Product(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    price: float
    currency: str = "USD"
    images: List[str] = []
    variants: List[ProductVariant] = []
    category: Optional[str] = None
    tags: List[str] = []
    is_active: bool = True
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


OrderStatus = Literal["pending", "confirmed", "processing", "shipped", "delivered", "cancelled", "refunded"]


class OrderItem(BaseModel):
    product_id: str
    variant_id: Optional[str] = None
    quantity: int
    unit_price: float
    total_price: float


class Order(BaseModel):
    id: str
    buyer_id: str
    seller_id: Optional[str] = None
    items: List[OrderItem]
    status: OrderStatus
    total_amount: float
    currency: str = "USD"
    shipping_address: Optional[Dict[str, Any]] = None
    notes: Optional[str] = None
    created_at: Optional[str] = None
    updated_at: Optional[str] = None


class CreateOrderInput(BaseModel):
    items: List[Dict[str, Any]]  # [{ product_id, variant_id?, quantity }]
    shipping_address: Optional[Dict[str, Any]] = None
    notes: Optional[str] = None
